packages/dev/core/src/Meshes/mesh.ts

```diff
@@ -1497,6 +1522,19 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
         this._instanceDataStorage.overridenInstanceCount = count;
     }
 
+    /** @internal */
+    public _getInstanceDataStorage(): _InstanceDataStorageRenderPass {
+        const renderPassId = this._instanceDataStorage.engine.isWebGPU ? this._instanceDataStorage.engine.currentRenderPassId : 0;

```

Comment:
This looks heavier than before I wonder how we could prevent the perf hit for every mesh of a scene as it should only change per render pass in theory ?